lib/services/collections_service.dart

```diff
@@ -308,18 +313,47 @@ class CollectionsService {
   }
 
   void _cacheCollectionAttributes(Collection collection) {
+    _collectionIDToCollections[collection.id] = collection;
+    final updatedCollection = getCollectionWithDecryptedName(collection);
```

Comment:
`updatedCollection` sounds misleading here. Call it `collectionWithDecryptedName` instead? Or forget about extracting it into a variable and directly do a 
```dart
_collectionIDToCollections[collection.id] = getCollectionWithDecryptedName(collection);
```